File: cases/taint/javascript/object-separation-positive/ObjectSeparationPositive.js
function dfb_source() { // DFB-SOURCE: object-separation-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: object-separation-sink

function run() {
  const tainted = { value: 0 };
  const clean = { value: 0 };
  tainted.value = dfb_source(); // DFB-WITNESS: object-separation-store
  clean.value = "clean";
  dfb_sink(tainted.value);
}

File: cases/taint/scala/object-separation-positive/ObjectSeparationPositive.scala
package dataflowbench

object ObjectSeparationPositive {
  class Holder {
    var value: Int = 0
  }

  def dfb_source(): Int = { // DFB-SOURCE: object-separation-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: object-separation-sink

  def run(): Unit = {
    val tainted = new Holder()
    val clean = new Holder()
    tainted.value = dfb_source() // DFB-WITNESS: object-separation-store
    clean.value = 0
    dfb_sink(tainted.value)
  }
}
